### DISCUSSION TITLE
yarpdatadumper with png files

### DISCUSSION THREAD

**Discussion by triccyx** [robotology/yarp#2537] [Support - Q&A] [answered]
**Describe the bug**
Using yarpdatadumper can not save images in png format. While bmp format works (--image)

**To Reproduce**
```bash
yarpserver --write
yarpdev --device grabberDual --subdevice usbCamera --camModel ultrapython --d /dev/media0 --name /grabber
yarpdatadumper --name /log --rxTime --txTime --type image_png --connect /grabber
```

**Expected behavior**
Inside folder log I can find the dumped images. No images found.

**Screenshots**
na

**Configuration (please complete the following information):**
 - OS:Ubuntu 20.04
 - yarp version:YARP version 3.4.101+20210309.4+gitae65e3050+dirty
-  compiler:gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

**Additional context**
Errors issued repeatedly:
```
libpng warning: Invalid bit depth in IHDR
libpng error: Invalid IHDR data
[ERROR]/home/triccyx/Documents/robotology-superbuild/robotology/YARP/src/libYARP_sig/src/yarp/sig/ImageFile.cpp:316 bool SavePNG(char*, const char*, size_t, size_t, size_t, png_byte, png_byte): [write_png_file] Error during writing header
libpng warning: Invalid bit depth in IHDR
libpng error: Invalid IHDR data
[ERROR]/home/triccyx/Documents/robotology-superbuild/robotology/YARP/src/libYARP_sig/src/yarp/sig/ImageFile.cpp:316 bool SavePNG(char*, const char*, size_t, size_t, size_t, png_byte, png_byte): [write_png_file] Error during writing header
```
Important notes:
- Try with the new UltraPython camera
- yarpdev on ARM arch, yarpdatadumper on X86 or Arm has the same result

**Comment by pattacini** [comment, marked as answer]
By launching:
```console
$ yarpdev --device fakeFrameGrabber &
$ yarpdatadumper --connect /grabber --type image_png
```
I managed to get the PNG images stored correctly.

Could you do the same test on your system @triccyx.

**Comment by triccyx** [comment]
I have tried on the ARM Enclustra board and all seems ok. The files are generated. No errors detected.

**Reply by triccyx** [reply]
to clarify: I have tested with @pattacini bash commands and it works. But it doesn't work with the real camera

**Reply by pattacini** [reply]
`yarpdatadumper` calls the library services `SavePNG()` developed by @randaz81.
Maybe he can provide some insights.

**Reply by triccyx** [reply]
I have tried with jpg with success.
```
yarpdatadumper --name /log --rxTime --txTime --type image_jpg --connect /grabber
```

**Reply by randaz81** [reply]
@triccyx  I suspect that the error you see is due to the fact that your image is not a standard rgb image and libpng does not know how to handle it. Could you provide more details on the image format or give me access to your camera for a check? Anyway, looks like this discussion will become a standard yarp issue (bug/feature request).

PS: the difference between png and jpg is the following: png is lossless.

**Reply by pattacini** [reply]
>Anyway, looks like this discussion will become a standard yarp issue (bug/feature request).

Indeed it was and then was converted into a discussion – I don't know why.

**Reply by triccyx** [reply]
> @triccyx I suspect that the error you see is due to the fact that your image is not a standard rgb image and libpng does not know how to handle it. Could you provide more details on the image format or give me access to your camera for a check? Anyway, looks like this discussion will become a standard yarp issue (bug/feature request).
> 
> PS: the difference between png and jpg is the following: png is lossless.

Yes, you are welcome when you can.

**Comment by support** [comment]
:wave: @triccyx, we use the issue tracker exclusively for bug reports and feature requests. However, this issue appears to be a question, a setup issue, or a support request. Please use our [`robotology/community` repository](https://github.com/robotology/community/discussions/) to get help next time. This issue will be moved to the correct issue tracker by a human ASAP, so that you can get proper support.
